### PR TITLE
support for CGMES 3 (CIM version 100)

### PR DESCRIPTION
**Pull request by zamarrenolm**
Signed-off-by: Luma <zamarrenolm@aia.es>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Comment by annetill** [comment]
In a previous meeting, I have noted some key points for the main changes we will have with version 3.0:
```
- TP confirmed as output; no Boundary TP profile. 
- Model are always built with Connectivity Nodes.
- 1 TP instance for a merged model
- MAS declaration in header required for merged SV, DL and TP.
- Header dependency revised.
- Schedules moved to CoreEquipment; Operation: only measurement classes remain.
- Separate RDFS for CoreEquipment, Operation and ShortCircuit
```
etc. I can share the slides but it is not the final version. What I understand is that bus/branch topology is not supported in version 3.0, DL and TP profils becomes outputs. Do you have planned to do that in that PR for TP profile?

**Comment by zamarrenolm** [comment]
> In a previous meeting, I have noted some key points for the main changes we will have with version 3.0:
> 
> ```
> - TP confirmed as output; no Boundary TP profile. 
> - Model are always built with Connectivity Nodes.
> - 1 TP instance for a merged model
> - MAS declaration in header required for merged SV, DL and TP.
> - Header dependency revised.
> - Schedules moved to CoreEquipment; Operation: only measurement classes remain.
> - Separate RDFS for CoreEquipment, Operation and ShortCircuit
> ```
> 
> etc. I can share the slides but it is not the final version. What I understand is that bus/branch topology is not supported in version 3.0, DL and TP profils becomes outputs. Do you have planned to do that in that PR for TP profile?

I would consider the TP export in a separate PR.

With the contents of this PR we are able to successfully import the available samples for the new format.

File: cgmes/cgmes-model/src/main/resources/CIM100.sparql [hunk 1]
```diff
@@ -30,3 +30,63 @@ OPTIONAL { GRAPH ?grapBDTP {
...
+        cim:IdentifiedObject.name ?name ;
+        cim:RotatingMachine.GeneratingUnit ?GeneratingUnit .
+    OPTIONAL {
+        ?SynchronousMachine cim:RotatingMachine.ratedS ?ratedS
+       }
+    ?GeneratingUnit
+        a ?generatingUnitType ;
+        cim:GeneratingUnit.minOperatingP ?minP ;
+        cim:GeneratingUnit.maxOperatingP ?maxP ;
+        cim:GeneratingUnit.nominalP ?initialP ;
```
Comment (zamarrenolm) [inline cgmes/cgmes-model/src/main/resources/CIM100.sparql:48]: initialP attribute has been removed from EQ in CIM100: _This required attribute is deleted as there is a general agreement that it is not expected to be able to run power flow on the EQ alone. SSH is required to be able to execute power flow._ 

So previously a required attribute, now can be dropped entirely, as initial values should come from SSH. In the conversion, a default value of zero should be set if no data from SSH is read.

File: cgmes/cgmes-model/src/main/resources/CIM100.sparql [hunk 2]
```diff
@@ -30,3 +30,63 @@ OPTIONAL { GRAPH ?grapBDTP {
...
+# query: operationalLimits
+SELECT *
+WHERE
+{ GRAPH ?graph {
+    ?OperationalLimit
+        a $OperationalLimitSubclass ;
+        cim:IdentifiedObject.name ?name ;
+        cim:OperationalLimit.OperationalLimitSet ?OperationalLimitSet ;
+        cim:OperationalLimit.OperationalLimitType ?OperationalLimitType .
+    OPTIONAL { ?OperationalLimit cim:CurrentLimit.normalValue ?value }
```
Comment (zamarrenolm) [inline cgmes/cgmes-model/src/main/resources/CIM100.sparql:76]: In CIM100 the attribute `normalValue` comes from EQ. There is also an attribute `value` defined now in the SSH profile. We could assume that values read from SSH should overwrite values obtained from EQ.

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/OperationalLimitConversion.java [hunk 3]
```diff
@@ -192,7 +192,8 @@ public boolean valid() {
 
     @Override
     public void convert() {
-        double value = p.asDouble("value");
+        double normalValue = p.asDouble("normalValue");
+        double value = p.asDouble("value", normalValue);
```
Comment (annetill) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/OperationalLimitConversion.java:196]: I don't understand the conversion here. The limits could be in units or in percentages. In the second case, we need the normal value to convert the limit in unit in IIDM. Where have we that conversion?

File: cgmes/cgmes-conformity/src/test/java/com/powsybl/cgmes/conformity/test/Cgmes3Catalog.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    public static TestGridModelResources microGrid() {
+        String base = CGMES_3_TEST_MODELS + "/MicroGrid/";
+        String baseBoundary = base;
+        return new TestGridModelResources(
+                "MicroGrid",
+                null,
+                new ResourceSet(base, "" +
+                    "20210209T1930Z_1D_ASSEMBLED_DL_9.xml",
```
Comment (annetill) [inline cgmes/cgmes-conformity/src/test/java/com/powsybl/cgmes/conformity/test/Cgmes3Catalog.java:28]: Are you sure that as inputs `20210209T1930Z_1D_ASSEMBLED_DL_9`, `20210209T1930Z_1D_ASSEMBLED_SV_9`, `20210209T2323Z_1D_ASSEMBLED_TP_9` should be imported as the other files? It should be outputs.

File: cgmes/cgmes-conformity/src/test/java/com/powsybl/cgmes/conformity/test/Cgmes3Catalog.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    public static TestGridModelResources miniGrid() {
+        String base = CGMES_3_TEST_MODELS + "/MiniGrid/";
+        String baseBoundary = base;
+        return new TestGridModelResources(
+                "MiniGrid",
+                null,
+                new ResourceSet(base, "" +
+                    "20210202T1930Z_1D_ASSEMBLED_DL_7.xml",
```
Comment (annetill) [inline cgmes/cgmes-conformity/src/test/java/com/powsybl/cgmes/conformity/test/Cgmes3Catalog.java:47]: Same remark as before: `ASSEMBLED` files should be removed for me, and the `TP` file should be treated as an output too.

File: cgmes/cgmes-conformity/src/test/java/com/powsybl/cgmes/conformity/test/Cgmes3Catalog.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+                new ResourceSet(baseBoundary, "MiniGridTestConfiguration_EQ_BD_v3.0.0.xml"));
+    }
+
+    public static TestGridModelResources smallGrid() {
+        String base = CGMES_3_TEST_MODELS + "/SmallGrid/";
+        String baseBoundary = base;
+        return new TestGridModelResources(
+                "SmallGrid",
+                null,
+                new ResourceSet(base, "" +
```
Comment (annetill) [inline cgmes/cgmes-conformity/src/test/java/com/powsybl/cgmes/conformity/test/Cgmes3Catalog.java:61]: Same remark for `TP` file. It is what I have understand from the CGMES 3.0: it is becoming an output. Maybe to be done in another PR. If so, put please a FIXME and a comment to remember.

File: cgmes/cgmes-conformity/src/test/java/com/powsybl/cgmes/conformity/test/Cgmes3Catalog.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        String base = CGMES_3_TEST_MODELS + "/Svedala/";
+        return new TestGridModelResources(
+                "Svedala",
+                null,
+                new ResourceSet(base, "" +
+                    "20201202T1843Z_1D_Svedala Area_DL_001.xml",
+                    "20201202T1843Z_1D_Svedala Area_SV_001.xml",
+                    "20201202T1843Z_1D_Svedala Area_EQ_001.xml",
+                    "20201202T1843Z_1D_Svedala Area_SSH_001.xml",
+                    "20201202T1843Z_1D_Svedala Area_TP_001.xml"));
```
Comment (annetill) [inline cgmes/cgmes-conformity/src/test/java/com/powsybl/cgmes/conformity/test/Cgmes3Catalog.java:81]: Same remark.

File: cgmes/cgmes-conformity/src/test/resources/cgmes3-test-models/MicroGrid/20171002T0930Z_ENTSO-E_EQ_BD_2.xml [hunk 8]
```diff
@@ -0,0 +1,269 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<rdf:RDF xmlns:cim="http://iec.ch/TC57/CIM100#" xmlns:eu="http://iec.ch/TC57/CIM100-European#" xmlns:md="http://iec.ch/TC57/61970-552/ModelDescription/1#" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
```
Comment (annetill) [inline cgmes/cgmes-conformity/src/test/resources/cgmes3-test-models/MicroGrid/20171002T0930Z_ENTSO-E_EQ_BD_2.xml:2]: This file is going to be deprecated. It means that all the infos should be in the `EQ` file. Is it true for the test files?

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/OperationalLimitConversion.java (hunk 3 above)
Comment (zamarrenolm) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/OperationalLimitConversion.java:196]: In CGMES 3 values for limits have been moved from EQ to SSH. EQ now contains `normalValue`, that should be used when no explicit `value` is defined in SSH. That is, `normalValue` in EQ is the default value for the limit if a value is not explicitly given in the SSH.

For Operational Limits, all the limit values and normal values are given in the corresponding engineering units, according to the specification. I think is for Measurements that there is the possibility of providing values in defined units or as percentages of normal values.

**Comment by zamarrenolm** [comment]
Net unit tests have been added that ensure CGMES 3 networks are properly imported even when the TP, SV files are not provided.

**Review by annetill — APPROVED** [review]
Thanks a lot for this relevant work.
